doc/rbd/rbd-snapshot.rst

```diff
@@ -206,107 +232,131 @@ performed these steps, you can begin cloning the snapshot.
 
 
 The cloned image has a reference to the parent snapshot, and includes the pool
-ID,  image ID and snapshot ID. The inclusion of the pool ID means that you may
-clone snapshots  from one pool to images in another pool.
-
-
-#. **Image Template:** A common use case for block device layering is to create a
-   master image and a snapshot that serves as a template for clones. For example, 
-   a user may create an image for a Linux distribution (e.g., Ubuntu 12.04), and 
-   create a snapshot for it. Periodically, the user may update the image and create
-   a new snapshot (e.g., ``sudo apt-get update``, ``sudo apt-get upgrade``,
-   ``sudo apt-get dist-upgrade`` followed by ``rbd snap create``). As the image
-   matures, the user can clone any one of the snapshots.
-
-#. **Extended Template:** A more advanced use case includes extending a template
-   image that provides more information than a base image. For example, a user may
-   clone an image (e.g., a VM template) and install other software (e.g., a database,
-   a content management system, an analytics system, etc.) and then snapshot the 
-   extended image, which itself may be updated just like the base image.
-
-#. **Template Pool:** One way to use block device layering is to create a 
-   pool that contains master images that act as templates, and snapshots of those
-   templates. You may then extend read-only privileges to users so that they 
-   may clone the snapshots without the ability to write or execute within the pool.
-
-#. **Image Migration/Recovery:** One way to use block device layering is to migrate
-   or recover data from one pool into another pool.
+ID, the image ID, and the snapshot ID. The inclusion of the pool ID means that
+you may clone snapshots from one pool to images in another pool.
+
+#. **Image Template:** A common use case for block device layering is to create
+   a base image and a snapshot that serves as a template for clones. For
+   example: a user may create an image for a Linux distribution (for example,
+   Ubuntu 22.04) and create a snapshot for it. The user may occasionally update
+   the image and create a new snapshot (by using such commands as ``sudo
+   apt-get update``, ``sudo apt-get upgrade``, or ``sudo apt-get dist-upgrade``
+   followed by ``rbd snap create``). As the image matures, the user can clone
+   any one of the snapshots.
+
+#. **Extended Template:** A more advanced use case includes extending a
+   template image that provides more information than a base image. For
+   example, a user may clone an image (for example, a VM template) and install
+   other software (for example, a database, a content management system, an
+   analytics system) and then snapshot the extended image, which may itself be
+   updated just like the base image.
+
+#. **Template Pool:** One way to use block device layering is to create a pool
+   that contains both (1) base images that act as templates and (2) snapshots
```

Comment:
This needs to be reworded.  A snapshot is always attached to the image, a case where an image lives an one pool and its snapshot lives in another pool is not possible so "(1) ... and (2) ..." doesn't make sense from a technical perspective.  I'd simplify to "a pool that contains base images that act as templates and respective snapshots".

Reply:
> This needs to be reworded. A snapshot is always attached to the image, a case where an image lives an one pool and its snapshot lives in another pool is not possible so "(1) ... and (2) ..." doesn't make sense from a technical perspective. I'd simplify to "a pool that contains base images that act as templates and respective snapshots".

@idryomov I've written it thus:

**Template Pool:** One way to use block device layering is to create a pool
   that contains both base images that act as templates and their associated
   snapshots. You may then extend read-only privileges to users so that they
   may clone the snapshots without the ability to write or execute within the
   pool.

Does this properly express what it should?

Reply:
I don't like "both" there.  One can't have base images without snapshots so "both" seems a bit redundant and misleading.  I'd either just drop "both" or replace "both" with "only" (or "just") to underscore that user-cloned images would need to go to a different pool as the point of the template pool is that it's read-only.

Reply:
@idryomov:

I've rewritten it thus:

> #. **Template Pool:** One way to use block device layering is to create a pool
>    that contains (1) base images that act as templates and (2) snapshots of
>    those templates. You may then extend read-only privileges to users so that
>    they may clone the snapshots even though they do not have permissions that
>    allow them to write or execute within the pool.

Let me know what you think.

Reply:
LGTM

Reply:
Qapla.